### ISSUE TITLE
Auth Challenge signing issue

### ISSUE THREAD

**Issue by dmgmyza** [Soneso/stellar-ios-mac-sdk#89]
It seems there is an issue with Auth Challenge signing (SEP-10) after upgrading SDK to version 1.8.0.
Signature created by SDK to sign the transaction is not valid.

For example, we signing next XDR:

```
https://laboratory.stellar.org/#xdr-viewer?input=AAAAAAZHmUf2xSOqrDLf0wK1KnpKn9gLAyk3Djc7KHL5e2YuAAAAAf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FAAAAAQAAAABe1CIaAAAAAF7UI0YAAAAAAAAAAQAAAAEAAAAA305u1W%2BC8ChCMyOCQa%2FOjrYFXs3VQvneddHTq%2Bp6CqcAAAAKAAAAClZhdWx0IGF1dGgAAAAAAAEAAABANGQ5ZjQ5OWNmMWE5ZTJiM2RkZWUyMWNjZGNmZjQ3MTIzZjgwM2UzNjdmZDYxZmY5Mjc1NGZmMTJhMWNmOWE0ZAAAAAAAAAAB%2BXtmLgAAAEAidHX75sVl7ZdXrkOL%2BEX7qskl%2F9xVMKkXC4lr1zjQQbNZyeO9Sa49BC1ln54k9FFvabWG0RAf7IChg4E7QN8C&type=TransactionEnvelope&network=public
```

We're using the next SDK methods to sign tx:

```
    let pk = "GDPU43WVN6BPAKCCGMRYEQNPZ2HLMBK6ZXKUF6O6OXI5HK7KPIFKO7CW"
    let seed = "SA6QS22REFMONMF3O7MMUCUVCXQIS6EHC63VY4FIEIF4KGET4BR6UQAI"
    let keyPair = try! KeyPair(secretSeed: seed)
    
    let transaction = "AAAAAAZHmUf2xSOqrDLf0wK1KnpKn9gLAyk3Djc7KHL5e2YuAAAAAf//////////AAAAAQAAAABe1CIaAAAAAF7UI0YAAAAAAAAAAQAAAAEAAAAA305u1W+C8ChCMyOCQa/OjrYFXs3VQvneddHTq+p6CqcAAAAKAAAAClZhdWx0IGF1dGgAAAAAAAEAAABANGQ5ZjQ5OWNmMWE5ZTJiM2RkZWUyMWNjZGNmZjQ3MTIzZjgwM2UzNjdmZDYxZmY5Mjc1NGZmMTJhMWNmOWE0ZAAAAAAAAAAB+XtmLgAAAEAidHX75sVl7ZdXrkOL+EX7qskl/9xVMKkXC4lr1zjQQbNZyeO9Sa49BC1ln54k9FFvabWG0RAf7IChg4E7QN8C"
      
    let transactionEnvelope = try! TransactionEnvelopeXDR(xdr: transaction)
    let txHash = try! [UInt8](transactionEnvelope.txHash(network: .public))
    transactionEnvelope.appendSignature(signature: keyPair.signDecorated(txHash))
```

Here is resulting XDR after signing tx with SDK:
```
AAAAAAZHmUf2xSOqrDLf0wK1KnpKn9gLAyk3Djc7KHL5e2YuAAAAAf//////////AAAAAQAAAABe1CIaAAAAAF7UI0YAAAAAAAAAAQAAAAEAAAAA305u1W+C8ChCMyOCQa/OjrYFXs3VQvneddHTq+p6CqcAAAAKAAAAClZhdWx0IGF1dGgAAAAAAAEAAABANGQ5ZjQ5OWNmMWE5ZTJiM2RkZWUyMWNjZGNmZjQ3MTIzZjgwM2UzNjdmZDYxZmY5Mjc1NGZmMTJhMWNmOWE0ZAAAAAAAAAAC+XtmLgAAAEAidHX75sVl7ZdXrkOL+EX7qskl/9xVMKkXC4lr1zjQQbNZyeO9Sa49BC1ln54k9FFvabWG0RAf7IChg4E7QN8C6noKpwAAAEA8fgrtV9ZYR3rmE3faiZzx0tG7ANpaUqG51vJIUJFDZ036cFONnCTIPA2XoXn19FmVywCsAqzj8O6Bum9/ejMO
```

Signature added by SDK:
PH4K7VfWWEd65hN32omc8dLRuwDaWlKhudbySFCRQ2dN+nBTjZwkyDwNl6F59fRZlcsArAKs4/Dugbpvf3ozDg==

Our real signature (was generated by Stellar Laboratory): TrYAEx9qMRnym42XNlZWJlshMCTJ1k1Adzc865ydc0OVWxymmAP+2MJcrkmgfgFZn5wcYYxzaWrQ4gCqYrlCDw==

The same code is working with version 1.7.4 (signatures from Laboratory and SDK are the same).

**Comment by christian-rogobete** [comment]
Hi @dmgmyza, I just released a new version 1.8.2 containing a fix for this issue. Your code should run correctly with this version. Please consider to use only V1 transactions as soon as protocol 13 is released in the main net because then we will remove support for protocol 12. Right now you can test it also like this:

```swift
let keyPair = try! KeyPair(secretSeed: "SA6QS22REFMONMF3O7MMUCUVCXQIS6EHC63VY4FIEIF4KGET4BR6UQAI")

let xdr = "AAAAAAZHmUf2xSOqrDLf0wK1KnpKn9gLAyk3Djc7KHL5e2YuAAAAAf//////////AAAAAQAAAABe1CIaAAAAAF7UI0YAAAAAAAAAAQAAAAEAAAAA305u1W+C8ChCMyOCQa/OjrYFXs3VQvneddHTq+p6CqcAAAAKAAAAClZhdWx0IGF1dGgAAAAAAAEAAABANGQ5ZjQ5OWNmMWE5ZTJiM2RkZWUyMWNjZGNmZjQ3MTIzZjgwM2UzNjdmZDYxZmY5Mjc1NGZmMTJhMWNmOWE0ZAAAAAAAAAAB+XtmLgAAAEAidHX75sVl7ZdXrkOL+EX7qskl/9xVMKkXC4lr1zjQQbNZyeO9Sa49BC1ln54k9FFvabWG0RAf7IChg4E7QN8C"
        
let transaction = try! Transaction(envelopeXdr: xdr)
try! transaction.sign(keyPair: keyPair, network: .public, coreProtocolVersion: 13)
let xdrEnvelope = try! transaction.encodedEnvelope(coreProtocolVersion: 13)

print(xdrEnvelope)
````
Please let me know if it works on your side.

**Comment by christian-rogobete** [comment]
Hello @cloudy-d ,

thank you again for reporting this issue back then. My name is Christian and I am the main developer and maintainer of the Stellar iOS SDK. To secure funding for the maintenance and improvement of the SDK I'm applying to this quarter's round of the Stellar Community Fund - Public Goods Award and they'd like to incorporate user signal into the decision-making process.

Would you mind leaving a comment under my SCF PR:
https://github.com/SCF-Public-Goods-Maintenance/scf-public-goods-maintenance.github.io/pull/97

If you could state the context of your usage in the LOBSTR Wallet and any feedback you have, that would be great!

One thing to know: the comment window is short, it already closes on Wednesday, July 29.

Would love to have your support!
